pkg/kubelet/kubelet.go

```diff
@@ -2922,6 +2921,14 @@ func (kl *Kubelet) ListenAndServeReadOnly(address net.IP, port uint) {
 	server.ListenAndServeKubeletReadOnlyServer(kl, kl.resourceAnalyzer, address, port, kl.containerRuntime)
 }
 
+func (kl *Kubelet) cleanUpContainersInPod(podId types.UID, exitedContainerID string) {
```

Comment:
The logic behind is a bit complex, some comments are appreciated. Why we can delete container here.